### DISCUSSION TITLE
Provide a GBytes API when possible

### DISCUSSION THREAD

**Discussion by tintou** [libvips/libvips#3080] [Ideas]
From an introspection perspective, the usage of raw pointers in vips_area and vips_blob makes it difficult to use the API, it would be great to be able to provide a GBytes-based API when possible as it already covers many use cases regarding the data handling.

**Comment by jcupitt** [comment]
Hi @tintou,

GBytes looks very similar to VipsArea, so I agree we could probably add API for this pretty easily. I suppose something to make each from the other would be a start -- it looks like it should be possible to avoid copies.

We could also (probably?) junk VipsArea and just replace it with GBytes, with a few simple compatibility functions for old code. Though that sounds more intrusive.

**Reply by jcupitt** [reply]
... I made an enhancement issue: https://github.com/libvips/libvips/issues/3081

Thanks for the suggestion!
